### DISCUSSION TITLE
Custom OIDC identity provider - GUI presents very limited fields

### DISCUSSION THREAD

**Discussion by kamilaprz** [keycloak/keycloak#35605] [Q&A]
Hello everyone, 

We are working with Keycloak v25.0.6.

We have developed custom identity providers in our system, extending `KeycloakOIDCIdentityProvider` and `KeycloakOIDCIdentityProviderFactory`.

However, we've noticed an issue when adding a new identity provider of these types in the Admin UI. The interface presents only a very limited set of fields:

* Redirect URI
* Client ID
* Client Secret
* Display Order

These custom providers are intended to have the same properties as `KeycloakOIDCIdentityProvider`, functioning as generic OIDC providers with additional functionality layered on top.

The IDs for these providers are as following convention:
* {custom-prefix}-oidc-{specific-suffix}
* {custom-prefix}-oidc

I looked briefly at those resources:
https://github.com/keycloak/keycloak/blob/main/js/apps/admin-ui/src/identity-providers/add/DetailSettings.tsx#L406

and
https://github.com/keycloak/keycloak/blob/main/js/apps/admin-ui/src/identity-providers/add/AddOpenIdConnect.tsxL82

I would expect, at the very least, the ability to edit the *alias* for a new provider.

Here is an example code snippet for the custom identity provider implementation:
```java
@MetaInfServices(IdentityProviderFactory.class)
public class CustomOidcProviderFactory extends KeycloakOIDCIdentityProviderFactory {
    public static final String PROVIDER_ID = "custom-oidc";
    private static final String PROVIDER_DESC = "Custom OpenID Connect";

    @Override
    public String getId() {
        return PROVIDER_ID;
    }

    @Override
    public String getName() {
        return PROVIDER_DESC;
    }

    @Override
    public KeycloakOIDCIdentityProvider create(KeycloakSession session, IdentityProviderModel model) {
        return new CustomOidcProvider(session, new OIDCIdentityProviderConfig(model));
    }
}

public class CustomOidcProvider extends KeycloakOIDCIdentityProvider {

    public CustomOidcProvider(KeycloakSession session, OIDCIdentityProviderConfig config) {
        super(session, config);
    }

    @Override
    public BrokeredIdentityContext getFederatedIdentity(String response) {
        // do something extra
        return context;
    }
}
```

This is problematic because we plan to have multiple identity providers of the same type. As you know, the alias for an identity provider must be unique, is included in the redirect URI, and *cannot be changed* after creation of new identity provider.

Is there something in our implementation or configuration that we might be missing to enable the full set of fields, including alias, to appear? Any insights would be greatly appreciated.

Thank you!

**Comment by Aldisti** [comment]
I was trying to create a new OIDC Provider that just added a query param to the authorize request.
Initially I did that by extending the OIDCIdentityProvider class but the generated GUI was different than the default OIDC.

In the end, I replaced the provider id **custom-oidc** with **oidc**. This way I got all the OIDC configurations, but I overrided the default OIDC provider.

But, I think this could be improved...
